internal/pkg/propagation/propagation.go

```diff
@@ -254,37 +252,10 @@ func (prop *Propagation) taintCall(call *ssa.Call, maxInstrReached map[*ssa.Basi
 		return
 	}
 
-	// Do not traverse through a method call if it is being reached via a Source receiver.
-	// Do traverse if the call is being reached through a tainted argument.
-	// Source methods that return tainted values regardless of their arguments should be identified by the fieldpropagator analyzer.
-	if recv := call.Call.Signature().Recv(); recv != nil && sourcetype.IsSourceType(prop.config, prop.taggedFields, recv.Type()) {
-		// If the receiver is not statically known (it has interface type) and the
-		// method has no arguments, Args will be empty.
-		if len(call.Call.Args) == 0 {
-			return
-		}
-		visitingFromArg := false
-		for i, a := range call.Call.Args {
-			// If the receiver's type is statically known,
-			// it will be the first element of the Args slice.
-			if !call.Call.IsInvoke() && i == 0 {
-				continue
-			}
-			if prop.tainted[a.(ssa.Node)] {
-				visitingFromArg = true
-			}
-		}
-		if !visitingFromArg {
-			// unmark the node so that it may be visited again when taint propagates to an argument
-			prop.tainted[call] = false
-			return
-		}
-	}
-
-	prop.taintReferrers(call, maxInstrReached, lastBlockVisited)
-	for _, a := range call.Call.Args {
-		prop.taintCallArg(a, maxInstrReached, lastBlockVisited)
+	if callee := call.Call.StaticCallee(); callee != nil {
```

Comment:
Can a `call` be both `StdlibCall` and `StdlibInterfaceCall`?

Reply:
Unless I've made a mistake, no. None of the functions in `funcSummaries` match the name/signature pairs in `interfaceFuncSummaries`.

Reply:
An explicit `if / else` or an early `return` in the `if` block would be warranted, then.  From a program-flow perspective, you're indicating that a thing could be both.